contrib/pylightning/lightning/plugin.py

```diff
@@ -306,9 +306,20 @@ def _getmanifest(self):
                 doc = "Undocumented RPC method from a plugin."
             doc = re.sub('\n+', ' ', doc)
 
+            argspec = inspect.getargspec(func)
+            args = argspec.args[1:]
+            defaults = argspec.defaults
+
+            # Make optional args be surrounded by square brackets
+            # list regular lightning-cli commands args
+            if defaults:
```

Comment:
Detecting default args below should still be correct unless someone tried to set a default for `plugin` or `request` like
```python
def hello(plugin, request=None, name="world')
```
In which case, you will know this for sure `if len(defaults) > len(args)`, in which case you could ditch method registration and print an error like "plugin and request args do not require default".

Reply:
This is fixed by my latest comment above.